pepper-apis/dss-core/src/main/java/org/broadinstitute/ddp/service/FileUploadService.java

```diff
@@ -285,6 +297,50 @@ public int removeUnusedUploads(Handle handle, Instant olderThanTimestamp, Intege
         return uploadIdsToDelete.size();
     }
 
+    public void sendNotifications(final Handle handle) {
+        StreamEx.of(handle.attach(FileUploadDao.class).findWithoutSentNotification())
+                .groupingBy(FileUpload::getStudyId)
+                .forEach((studyId, fileUploads) -> sendNotifications(handle, studyId, fileUploads));
+    }
+
+    private void sendNotifications(final Handle handle, final Long studyId, final List<FileUpload> fileUploads) {
+        final var study = handle.attach(JdbiUmbrellaStudy.class).findById(studyId);
+        if (StringUtils.isBlank(study.getNotificationEmail())) {
+            handle.attach(FileUploadDao.class).setNotificationSentByStudyId(studyId);
+            log.warn("Study {} doesn't have an e-mail for notifications", study.getGuid());
+            return;
+        }
+
+        StreamEx.of(fileUploads)
+                .groupingBy(FileUpload::getParticipantUserId)
+                .forEach((participantId, uploads) -> sendNotification(handle, study, participantId, uploads));
+
+        log.info("Notifications sent for {} study", study.getGuid());
+    }
+
+    private void sendNotification(final Handle handle,
+                                  final StudyDto study,
+                                  final Long participantId,
+                                  final List<FileUpload> fileUploads) {
+        final var result = sendGridClient.sendMail(new Mail(
+                new Email(study.getStudyEmail()),
+                "New files were uploaded in " + study.getName() + " study",
+                new Email(study.getNotificationEmail()),
+                new Content(
+                        "text/html",
+                        "User " + participantId + " uploaded following files: " + StreamEx.of(fileUploads)
```

Comment:
Do we want the internal participantId to be in the email to steady staff ? 

I believe we should include user "hruid" which is the usual mode of communication and that might help study managers searching for this participant in DSM.

Reply:
Will replace with hruid.